### ISSUE TITLE
TreeList - the getSelectedRowKeys method improvements

### ISSUE THREAD

**Issue by dxbykov** [DevExpress/DevExtreme#3209]
# The Problem 

Previously, the following tasks could not be easily performed while using recursive selection in the TreeList:

* Getting only the selected leaves   
* Getting only the selected parents without leaves  
* Getting the selected leaves and their parents  
 
Refer to [a workaround solution](https://isc.devexpress.com/Thread/WorkplaceDetails/T590425) if you are using DevExtreme v17.2 and earlier. 
  
# The Solution 

We have added a parameter to the `getSelectedRowKeys()` method which can accept the following string arguments:
 
* *all* – returns all the selected nodes including the selected parents’ children;
* *excludeRecursive* – returns all the selected nodes excluding the selected parents’ children;
* *leavesOnly* – only returns the selected leaves;

For instance, if the current selection is ...

``` 
[x] A 
   [x] B 
   [x] C 
      [x] D 
``` 

… then these arguments result in the following values: 

``` 
treeList.getSelectedRowKeys("excludeRecursive")  
// returns: A 

treeList.getSelectedRowKeys("all") 
// returns: A, B, C, D 

treeList.getSelectedRowKeys("leavesOnly") 
// returns: B, D 
``` 

# Try It And Give Us Feedback 

Follow the installation guide from the [release page](https://github.com/DevExpress/DevExtreme/releases) and leave us feedback on this topic. 

Feel free to leave comments or use [GitHub reactions](https://github.com/blog/2119-add-reactions-to-pull-requests-issues-and-comments).

**Comment by genachka** [comment]
:+1: Thank you for adding this!

**Comment by dxbykov** [comment]
Thanks @genachka and everybody who gave us early feedback by other means. The feature is now available in the [v18.1](https://github.com/DevExpress/DevExtreme/releases/tag/18.1.3) release. I'm closing this thread. In the case of bugs or questions, feel free to create new GitHub [issues](https://github.com/DevExpress/DevExtreme/issues) or tickets in our [Support Center](https://www.devexpress.com/Support/Center/).
